lib/operations/collection_ops.js

```diff
@@ -914,7 +1272,23 @@ function stats(coll, options, callback) {
   options = getReadPreference(coll, options, coll.s.db, coll);
 
   // Execute the command
-  coll.s.db.command(commandObject, options, callback);
+  executeCommand(coll.s.db, commandObject, options, callback);
+}
+
+function updateCallback(callback, err, r) {
```

Comment:
Nice!

Two things:

1. I'd switch the order of the parameters to be `err, r, callback`, since that is a common node convention.
2. I think that, since you are already checking for a callback at the top, you can remove the checks for the callback in `if (err && callback)` and `if (callback)`